scripts/tests/test_validate_docstrings.py

```diff
@@ -68,6 +87,23 @@ def sample(self):
         """
         return random.random()
 
+    @Decorators.good_decorator
```

Comment:
Can you use `@functools.lru_cache` instead, so we don't need to implement a custom decorator. I guess `lru_cache` is well wrapped.

When importing, I think it's a better practice to `import functools` than `from functools import wraps, lru_cache`.